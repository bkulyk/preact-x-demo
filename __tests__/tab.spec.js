import React from 'react';
import Tab from '@bitchin/tab';
import { shallow } from './enzyme';
import { it, expect } from './jest'

it('should render correctly', () => {
  expect(shallow((
    <Tab
      label="Nearby"
    />
  ))).toMatchSnapshot();
});

it('should render correctly when active', () => {
  expect(shallow((
    <Tab
      label="Favorite"
      active
      icon="favorite"
    />
  ))).toMatchSnapshot();
});

it('should render correctly when icon and label are stacked', () => {
  expect(shallow((
    <Tab
      label="Favorite"
      active
      icon="favorite"
      stacked
    />
  ))).toMatchSnapshot();
});
