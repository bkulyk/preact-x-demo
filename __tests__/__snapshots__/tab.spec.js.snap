// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<button
  aria-selected="true"
  className="mdc-tab"
  role="tab"
  tabIndex="0"
  type="button"
>
  <span
    className="mdc-tab__content"
  >
    <span
      className="mdc-tab__text-label"
    >
      Nearby
    </span>
  </span>
  <span
    className="mdc-tab-indicator"
  >
    <span
      className="mdc-tab-indicator__content mdc-tab-indicator__content--underline"
    />
  </span>
  <span
    className="mdc-tab__ripple"
  />
</button>
`;

exports[`should render correctly when active 1`] = `
<button
  aria-selected="true"
  className="mdc-tab mdc-tab--active"
  role="tab"
  tabIndex="0"
  type="button"
>
  <span
    className="mdc-tab__content"
  >
    <span
      aria-hidden="true"
      className="mdc-tab__icon material-icons"
    >
      favorite
    </span>
    <span
      className="mdc-tab__text-label"
    >
      Favorite
    </span>
  </span>
  <span
    className="mdc-tab-indicator mdc-tab-indicator--active"
  >
    <span
      className="mdc-tab-indicator__content mdc-tab-indicator__content--underline"
    />
  </span>
  <span
    className="mdc-tab__ripple"
  />
</button>
`;

exports[`should render correctly when icon and label are stacked 1`] = `
<button
  aria-selected="true"
  className="mdc-tab mdc-tab--stacked mdc-tab--active"
  role="tab"
  tabIndex="0"
  type="button"
>
  <span
    className="mdc-tab__content"
  >
    <span
      aria-hidden="true"
      className="mdc-tab__icon material-icons"
    >
      favorite
    </span>
    <span
      className="mdc-tab__text-label"
    >
      Favorite
    </span>
  </span>
  <span
    className="mdc-tab-indicator mdc-tab-indicator--active"
  >
    <span
      className="mdc-tab-indicator__content mdc-tab-indicator__content--underline"
    />
  </span>
  <span
    className="mdc-tab__ripple"
  />
</button>
`;
